File: runtime/ruby/units/rt/i32_rem_s.rb
# requires: rt/s32, rt/trap
def i32_rem_s(a, b)
  sa = s32(a)
  sb = s32(b)
  trap("integer divide by zero") if sb == 0
  r = sa.abs % sb.abs
  r = -r if sa < 0
  r & M32
end

File: runtime/python/units/rt/i32_rem_s.py
# requires: rt/s32, rt/trap
@staticmethod
def i32_rem_s(a, b):
    sa = Rt.s32(a)
    sb = Rt.s32(b)
    if sb == 0:
        Rt.trap("integer divide by zero")
    r = abs(sa) % abs(sb)
    if sa < 0:
        r = -r
    return r & Rt.M32
